pkg/addonmanager/controllers/addoninstall/controller.go

```diff
@@ -119,10 +119,11 @@ func (c *addonInstallController) sync(ctx context.Context, syncCtx factory.SyncC
 }
 
 func (c *addonInstallController) applyAddon(ctx context.Context, addonName, clusterName, installNamespace string) error {
-	addon, err := c.managedClusterAddonLister.ManagedClusterAddOns(clusterName).Get(addonName)
-	switch {
-	case errors.IsNotFound(err):
-		addon = &addonapiv1alpha1.ManagedClusterAddOn{
+	_, err := c.managedClusterAddonLister.ManagedClusterAddOns(clusterName).Get(addonName)
```

Comment:
do we need to add the check if the managedCluster is deleting firstly?

Reply:
It is checked upfront already